src/elfpy/pricing_models/base.py

```diff
@@ -525,6 +529,7 @@ def get_max_short(
                     or market_state_post_trade.share_price * market_state_post_trade.share_reserves
                     < market_state_post_trade.base_buffer
                     or market_state_post_trade.bond_reserves < market_state_post_trade.bond_buffer
+                    or market_state_post_trade.share_reserves < 0
```

Comment:
Why are you adding this? Adding a test given the TODO above (line 520) makes me nervous. Do we need all of these conditionals? If you have it in your head, can you add a comment on what these conditionals are achieving? What's the logic summary for this block?

Reply:
this was when I thought there was an error with get_max_short (there isn't). I noticed it's not doing checks for share_reserves, so I added it 🤷

Reply:
since we don't know which of those checks are necessary anyway, adding another one doesn't change that. I'm not attempting to do a thorough check of whether these are necessary here. If leaving this additional check out makes you more comfortable, I can do that.

Reply:
that function has been refactored